spec/bolt/outputter/outputter_spec.rb

```diff
@@ -0,0 +1,181 @@
+    # This is just the result set
+    expect(outputter.stringify(plan_result))
+      .to eq(<<~PLAN_RESULT.chomp)
+      [
+        {
+          "target": "target1",
+          "action": "action",
+          "object": null,
+          "status": "success",
+          "value": {
+            "_output": "ok"
+          }
+        },
+        {
+          "target": "target2",
+          "action": "action",
+          "object": null,
+          "status": "failure",
+          "value": {
+            "_error": {
+              "msg": "oops"
+            }
+          }
+        }
+      ]
+    PLAN_RESULT
+  end
+
+  it "formats result sets" do
+    expect(outputter.stringify(result_set))
+      .to eq(<<~RESULT_SET.chomp)
+      [
+        {
+          "target": "target1",
+          "action": "action",
+          "object": null,
+          "status": "success",
+          "value": {
+            "_output": "ok"
+          }
+        },
+        {
+          "target": "target2",
+          "action": "action",
+          "object": null,
+          "status": "failure",
+          "value": {
+            "_error": {
+              "msg": "oops"
+            }
+          }
+        }
+      ]
+    RESULT_SET
+  end
+
+  it "formats a result" do
+    expect(outputter.stringify(result))
+      .to eq(<<~RESULT.chomp)
+      {
+        "target": "target1",
+        "action": "action",
+        "object": null,
+        "status": "success",
+        "value": {
+          "_output": "ok"
+        }
+      }
+    RESULT
+  end
+
+  it "formats an apply result" do
+    expect(outputter.stringify(apply_result))
+      .to eq(<<~APPLY_RESULT.chomp)
+    {
+      "target": "target1",
+      "action": "apply",
+      "object": null,
+      "status": "success",
+      "value": {
+        "report": {
+          "status": "changed"
+        }
+      }
+    }
+    APPLY_RESULT
+  end
+
+  it "formats resource instances" do
+    expect(outputter.stringify(resource))
+      .to eq("File[/etc/puppetlabs/]")
+  end
+
+  it "formats errors" do
+    expect(outputter.stringify(error))
+      .to eq("Task 'watermelon' could not be found")
+  end
+
+  it "formats targets" do
+    expect(outputter.stringify(target))
+      .to eq("target1")
+  end
+
+  it "formats arrays of complex objects" do
+    expect(outputter.stringify([target, result_set, ['subarray']]))
+      .to eq(<<~ARRAY.chomp)
+   [
+     "target1",
+     [
+       {
+         "target": "target1",
+         "action": "action",
+         "object": null,
+         "status": "success",
+         "value": {
+           "_output": "ok"
+         }
+       },
+       {
+         "target": "target2",
+         "action": "action",
+         "object": null,
+         "status": "failure",
+         "value": {
+           "_error": {
+             "msg": "oops"
+           }
+         }
+       }
+     ],
+     [
+       "subarray"
+     ]
+   ]
+   ARRAY
+  end
+
+  it "formats hashes of complex objects" do
+  end
```

Comment:
?

Reply:
Woops! I'll add this later this morning.